src/jdk.net/windows/native/libextnet/WindowsSocketOptions.c

```diff
@@ -90,18 +74,13 @@ JNIEXPORT void JNICALL Java_jdk_net_WindowsSocketOptions_setIpDontFragment0
 /*
  * Class:     jdk_net_WindowsSocketOptions
  * Method:    getIpDontFragment0
- * Signature: (I)Z;
+ * Signature: (IZ)Z;
  */
 JNIEXPORT jboolean JNICALL Java_jdk_net_WindowsSocketOptions_getIpDontFragment0
-(JNIEnv *env, jobject unused, jint fd) {
+(JNIEnv *env, jobject unused, jint fd, jboolean isIPv6) {
     int optval, rv, sz = sizeof(optval);
-    jint family = socketFamily(fd);
-    if (family == -1) {
-        handleError(env, family, "get socket family failed");
-        return JNI_FALSE;
-    }
 
-    if (family == AF_INET) {
+    if (!isIPv6) {
```

Comment:
In passing, it might be easier to read if we change this code to use `if (isIPv6) { ... } then { ...}`. Not important here to do as part of this PR, just an observation.